plugins/gimp/file-jxl-save.cc

```diff
@@ -10,122 +10,368 @@
 #undef MIN
 #undef CLAMP
 
-#include "lib/jxl/alpha.h"
+#include <glib/gi18n.h>
+#include <libgimp/gimp.h>
+#include <libgimp/gimpui.h>
+//#include <glib/gstdio.h>
+
+#include <string>
+#include <vector>
+
+#include "jxl/encode.h"
+#include "jxl/encode_cxx.h"
+#include "jxl/resizable_parallel_runner.h"
+#include "jxl/resizable_parallel_runner_cxx.h"
 #include "lib/jxl/base/file_io.h"
-#include "lib/jxl/base/thread_pool_internal.h"
-#include "lib/jxl/enc_file.h"
-#include "plugins/gimp/common.h"
+
+#define PLUG_IN_BINARY "file-jxl"
+#define SAVE_PROC "file-jxl-save"
 
 namespace jxl {
 
-namespace {
-
-template <bool has_alpha, size_t alpha_bits = 16>
-Status ReadBuffer(const size_t xsize, const size_t ysize,
-                  const std::vector<float>& pixel_data, PaddedBytes icc,
-                  CodecInOut* const io) {
-  constexpr float alpha_multiplier = has_alpha ? 1.f / 255 : 0.f;
-  Image3F image(xsize, ysize);
-  ImageF alpha;
-  if (has_alpha) {
-    alpha = ImageF(xsize, ysize);
-  }
-  const float* current_sample = pixel_data.data();
-  for (size_t y = 0; y < ysize; ++y) {
-    float* rows[3];
-    for (size_t c = 0; c < 3; ++c) {
-      rows[c] = image.PlaneRow(c, y);
-    }
-    float* const alpha_row = has_alpha ? alpha.Row(y) : nullptr;
-    for (size_t x = 0; x < xsize; ++x) {
-      for (float* const row : rows) {
-        row[x] = BufferFormat<GIMP_PRECISION_FLOAT_GAMMA>::ToFloat(
-            *current_sample++);
-      }
-      if (has_alpha) {
-        alpha_row[x] = alpha_multiplier *
-                       BufferFormat<GIMP_PRECISION_FLOAT_GAMMA>::ToFloat(
-                           *current_sample++);
-      }
-    }
-  }
+typedef struct {
+  gboolean tmp;
+} JpegXlSaveVals;
 
-  ColorEncoding color_encoding;
-  JXL_RETURN_IF_ERROR(color_encoding.SetICC(std::move(icc)));
-  io->metadata.m.color_encoding = color_encoding;
-  io->SetFromImage(std::move(image), color_encoding);
-  if (has_alpha) {
-    io->metadata.m.SetAlphaBits(alpha_bits);
-    io->Main().SetAlpha(std::move(alpha), /*alpha_is_premultiplied=*/false);
-  }
-  return true;
+static JpegXlSaveVals jxl_vals = {
+    TRUE  // tmp
+};
+
+bool save_dialog(void) {
```

Comment:
Likewise, `(void)` is a remnant from C:

```suggestion
bool save_dialog() {
```